modules/asm/main.tf

```diff
@@ -34,14 +34,16 @@ data "google_container_cluster" "asm" {
 }
 
 resource "kubernetes_namespace" "system" {
-  count = var.create_system_namespace ? 1 : 0
+  count = var.create_system_namespace && var.mesh_management != "MANAGEMENT_AUTOMATIC" ? 1 : 0
 
   metadata {
     name = "istio-system"
   }
 }
 
 resource "kubernetes_config_map" "asm_options" {
+  count = var.mesh_management != "MANAGEMENT_AUTOMATIC" ? 1 : 0
```

Comment:
With TF 1.1 (perhaps earlier as well), when adding `count` to an existing resource, Terraform will automatically move state to index 0, so these should be fine.